### PR TITLE
Extract InternalReportDelegate class from Client

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Extracts the logic for sending an internal bugsnag report from the `Client` to its own class.

There should be no functional change here, but this enables the addition of a unit test to verify the behaviour of internal reporting on top of existing E2E tests. It also reduces the responsibilities of the large `Client` class.

### CODE REVIEW

File: bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal data class ImmutableConfig(
 
     companion object {
         private const val HEADER_API_PAYLOAD_VERSION = "Bugsnag-Payload-Version"
-        private const val HEADER_API_KEY = "Bugsnag-Api-Key"
+        internal const val HEADER_API_KEY = "Bugsnag-Api-Key"
```
Comment (tomlongridge) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt]: I'm not sure why this change is required.

Comment (fractalwrench) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt]: `HEADER_API_KEY` is now accessed from the `InternalReportDelegate` rather than using a string constant. Because this is in another class it needs to be `internal` rather than `private` to be accessed.

Comment (fractalwrench) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/ImmutableConfig.kt]: The usage of this constant within `InternalReportDelegate` seemed to have escaped the changeset - I've updated this to replace the string constant with `HEADER_API_KEY`

File: bugsnag-android-core/src/main/java/com/bugsnag/android/InternalReportDelegate.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                @Override
+                public void run() {
+                    try {
+                        Delivery delivery = immutableConfig.getDelivery();
+                        DeliveryParams params = immutableConfig.errorApiDeliveryParams();
+
+                        // can only modify headers if DefaultDelivery is in use
+                        if (delivery instanceof DefaultDelivery) {
+                            Map<String, String> headers = params.getHeaders();
+                            headers.put("Bugsnag-Internal-Error", "true");
```
Comment (tomlongridge) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/InternalReportDelegate.java]: Can you extract this to a header in this class for consistency?

Comment (fractalwrench) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/InternalReportDelegate.java]: Extracted to a header in `ImmutableConfig`.
